aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1135,8 +1135,16 @@ inline void gpu_reduce_kernel(TensorIterator& iter, const ops_t& ops, ident_t id
 
   using traits = function_traits<decltype(&ops_t::reduce)>;
   using arg_t = typename traits::template arg<0>::type;
+  // at::Half/at::ComplexHalf overflows easily as it's range is very small.
+  // So when scalar_t and out_scalar_t are at::Half/at::ComplexHalf, we
+  // set can_accumulate_in_output to False.
+  static constexpr bool is_inp_out_type_half_or_chalf =
```

Comment:
this should include bfloat also. Even though bfloat range is enough, the intermediate truncation will happen.

Reply:
Have addressed the review :)